here-naksha-lib-view/src/main/java/com/here/naksha/lib/view/concurrent/ParallelQueryExecutor.java

```diff
@@ -116,9 +122,20 @@ private <FEATURE, CODEC extends FeatureCodec<FEATURE, CODEC>> Stream<ViewLayerRo
 
     try (MutableCursor<FEATURE, CODEC> cursor =
         session.execute(clonedRequest).mutableCursor(codecFactory)) {
+      featureCnt = cursor.asList().size();
       return cursor.asList().stream().map(row -> new ViewLayerRow<>(row, layerPriority, layer));
     } catch (NoCursor e) {
+      status = "NOK";
       throw unchecked(e);
+    } finally {
+      log.info(
+          "[View Request stats => streamId,layerId,method,status,timeTakenMs,fCnt] - ViewReqStats {} {} {} {} {} {}",
+          NakshaContext.currentContext().getStreamId(),
+          layer.getCollectionId(),
```

Comment:
can we assign `layer.getCollectionId()` to a final variable and then use that variable at all the places? (trying to avoid calling it multiple times from the same function)

Reply:
Fixed It.